apmapf.py

```diff
@@ -40,51 +44,80 @@
     Not,
     Implies,
     Equals)
+
 from pysmt.typing import INT
+
+# Return to showing errors again.
 sys.stderr = sys.__stderr__
-from bfs import BFS
 
+from bfs import BFS
 
+# Absolute value of x = Ite(x >= 0, x, -x) = if x >= 0 then x else -x
 def Abs(x):
     return Ite(x >= 0, x, -x)
 
-
+# c0, c1 are "adjacent" iff d_x(c0, c1) + d_y(c0, c1) <= 1.
+# That is, if their Manhattan distance is <= 1. 
 def Adj(c0, c1):
     return 1 >= Abs(c1[0] - c0[0]) + Abs(c1[1] - c0[1])
 
-
+# coord is on the grid iff its x, y coordinates are bounded below by the origin
+# (0, 0), and above by the width = height of the (square) grid.
 def IsOnGrid(coord, gridsize):
     return And(
         coord[0] >= 0,
         coord[1] >= 0,
         coord[0] < gridsize,
         coord[1] < gridsize)
 
-
+# The path is connected iff each point in the path is adjacent to its
+# predecessor (if it has one) and successor (if it has one).
 def IsConnected(path):
     return And([Adj(path[i-1], path[i]) for i in range(1, len(path))])
 
-
+# c0 is literally the same position as c1 if their x and y coordinates are both
+# equal.
 def SamePosition(c0, c1):
     return And(Equals(c0[0], c1[0]), Equals(c0[1], c1[1]))
 
-
+# IntifyCoords takes a vector art in (R^2)^n and returns the closest vector to
+# it in (Z^2)^n.
```

Comment:
The input is in (Z^2)^n, it's just that pysmt uses its own wrapper for constants, in this case `Int` (imported from `pysmt.shortcuts`).

Reply:
Gotcha, updated the comment accordingly, thanks.